pkg/informers/informers.go

```diff
@@ -0,0 +1,180 @@
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/tools/cache"
+)
+
+type GenericInformer interface {
+	Informer() cache.SharedIndexInformer
+	Lister() cache.GenericLister
+}
+
+// InformerFactoriesForNamespaces is a simple way to combine several shared informers into a single struct with unified listing power
+type InformerFactoriesForNamespaces interface {
+	ForResource(namespace string, resource schema.GroupVersionResource) (GenericInformer, error)
+	Namespaces() sets.String
+}
+
+type InformersForResource struct {
+	informers []GenericInformer
+}
+
+func NewInformersForResource(ifs InformerFactoriesForNamespaces, resource schema.GroupVersionResource) (*InformersForResource, error) {
+	namespaces := ifs.Namespaces().List()
+	var informers []GenericInformer
+
+	for _, ns := range namespaces {
+		informer, err := ifs.ForResource(ns, resource)
+		if err != nil {
+			return nil, errors.Wrapf(err, "error getting informer for resource %v", resource)
+		}
+		informers = append(informers, informer)
+	}
+
+	return &InformersForResource{
+		informers: informers,
+	}, nil
+}
+
+func (w *InformersForResource) Start(stopCh <-chan struct{}) {
+	for _, i := range w.informers {
+		go i.Informer().Run(stopCh)
+	}
+}
+
+func (w *InformersForResource) GetInformers() []GenericInformer {
+	return w.informers
+}
+
+func (w *InformersForResource) AddEventHandler(handler cache.ResourceEventHandler) {
+	for _, i := range w.informers {
+		i.Informer().AddEventHandler(handler)
+	}
+}
+
+func (w *InformersForResource) AddEventHandlerWithResyncPeriod(handler cache.ResourceEventHandler, resyncPeriod time.Duration) {
+	for _, i := range w.informers {
+		i.Informer().AddEventHandlerWithResyncPeriod(handler, resyncPeriod)
+	}
+}
+
+func (w *InformersForResource) HasSynced() bool {
+	for _, i := range w.informers {
+		if !i.Informer().HasSynced() {
+			return false
+		}
+	}
+
+	return true
+}
+
+func (w *InformersForResource) List(selector labels.Selector) ([]runtime.Object, error) {
+	var ret []runtime.Object
+
+	for _, inf := range w.informers {
+		objs, err := inf.Lister().List(selector)
+		if err != nil {
+			return nil, err
+		}
+		ret = append(ret, objs...)
+	}
+
+	return ret, nil
+}
+
+func (w *InformersForResource) ListAllByNamespace(namespace string, selector labels.Selector, appendFn cache.AppendFunc) error {
+	for _, inf := range w.informers {
+		err := cache.ListAllByNamespace(inf.Informer().GetIndexer(), namespace, selector, appendFn)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (w *InformersForResource) Get(name string) (runtime.Object, error) {
+	var err error
+
+	for _, inf := range w.informers {
+		var ret runtime.Object
+		ret, err = inf.Lister().Get(name)
+
```

Comment:
let's cuddle the error setting and the error check

Reply:
fixed